pkg/oc/cli/admin/release/image_mapper.go

```diff
@@ -175,7 +177,54 @@ func NewImageMapperFromImageStreamFile(path string, input *imageapi.ImageStream,
 		}
 		references[tag.Name] = ref
 	}
-	return NewImageMapper(references)
+	imageMapper, err := NewImageMapper(references)
+	if err != nil {
+		return nil, err
+	}
+
+	// load all version values from the input stream, including any defaults, to perform
+	// version substitution in the returned manifests.
+	versions := make(map[string]string)
+	tagsByName := make(map[string][]string)
+	for _, tag := range input.Spec.Tags {
+		if _, ok := references[tag.Name]; !ok {
+			continue
+		}
+		value := tag.Annotations["io.openshift.build.versions"]
+		if len(versions) > 0 {
+			continue
+		}
+		items, err := parseComponentVersionsLabel(value)
+		if err != nil {
+			return nil, fmt.Errorf("input image stream has an had invalid version annotation for tag %q: %v", tag.Name, value)
+		}
+		for k, v := range items {
+			existing, ok := versions[k]
+			if ok {
+				versions[k] = v
+				if existing != v {
+					tagsByName[k] = append(tagsByName[k], tag.Name)
+				}
+			} else {
+				tagsByName[k] = append(tagsByName[k], tag.Name)
+			}
+		}
+	}
+	defaults, _ := parseComponentVersionsLabel(input.Annotations["io.openshift.build.versions"])
```

Comment:
any harm in returning errors ? Or we can never have errors here ?

Reply:
I'll return the error, I didn't have a good reason to eat it